TP-Sampler/src/app.js

```diff
@@ -666,97 +666,133 @@ function setup() {
 }
 
 export function getOptionsForMeasure (measure) {
-  // right now we only ever have one output option because we only have one device.
-  // once we chain multiple devices, we'll have to adjust this approach.
-  function createOutputOption (selectEl) {
-    const outputOption = document.createElement("option");
-    outputOption.value = deviceName;
-    outputOption.textContent = deviceName;
-    outputOption.selected = true;
-    selectEl.append(outputOption);
-  }
+  // get list of attributes from CODAP
+  codapCom.getAttributesFromTable().then((attrs) => {
+    const attrNames = attrs.map((attr) => attr.name);
+    codapCom.getAllItems().then((items) => {
+      let attrMap = {};
+      attrNames.forEach((attrName) => {
+        if (attrName === deviceName && !isCollector) {
+          attrMap[attrName] = [...new Set(variables)];
```

Comment:
is the `[...new Set(variables)]` there to filter out duplicate variable values?

Reply:
Yes, exactly